website/docs/help-and-support/troubleshooting-guide/widget-errors.md

```diff
@@ -1,38 +1,66 @@
 
-* **A large array or a huge JSON object** - You can relook at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
-* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either selected row / triggered row to bind.
+* **A large array or a huge JSON object** - You can re-look at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
+* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either the selected row / triggered row to bind.
 
 Once you have figured out the new structure for the data, head to the [source data](../../reference/widgets/json-form#source-data) field to make changes.
 
+## Query/API/JS object execution errors
+You may see below errors when execute an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) using a widget property.
+
+### Action can't be triggered
+You may see the error when binding an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) execution to a widget property that expects data, and can't be used to trigger an action. 
+
+#### Error message
+Action cannot be triggered from this field.
+
+**To-Do to add screenshot**
+
+#### Cause
+An **action** is referred to as an operation that could be performed by executing an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/). 
+
+:::info
+Action can only be performed by binding it to an [async](../../core-concepts/writing-code/workflows#async-fields) field.
+:::
+
+When you bind an action to a [sync](../../core-concepts/writing-code/workflows#sync-fields) field that only expects data, the `action cannot be triggered from this field` error is thrown.
+
+#### Solution
+You could resolve the error by invoking a `.data` property of an API/query/JS object. For example, you have an API ```getFullNameOfLoggedInUser``` that returns a user's full name, and you wish to add the user's name based on the logged-in user to show a welcome text, `Welcome! <LOGGED_IN_USER_NAME>`. You could bind the response generated by the API `getFullNameOfLoggedInUser` to a text widget by calling the `.data` property. You could add the below code snippet in a mustache (`{{}}`) to the text property of the text widget instead of `getFullNameOfLoggedInUser.run()`.
```

Comment:
I think it might help to add a little explanation to clarify why we use `.data`, added in after the first sentence of the **Solution** heading. (It might be hard for new users to understand *why* they need to do this). I put together a little example to show what I mean:


> \#\#\#\# Solution
> You could resolve the error by invoking the `.data` property of the API/query/JS object. Widgets' property fields are synchronous, and they can't be used to call queries (which are async). Instead, once a query has been run, the response is stored as an object on the `<query_name>.data` property. Widgets can accept and work with `<query_name>.data`, since it is just a simple object.
> 
> :::note
> Binding `<query_name>.data` to one of a widget's property fields will make that query run automatically on page load. (This can be switched off in the query's settings page).
> :::
> 
> For example, you have an API ```getFullNameOfLoggedInUser``` that returns a user's...

Reply:
@spark-c - As disccused, added the cross-reference for https://docs.appsmith.com/core-concepts/writing-code/workflows#use-case